qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/HandlerThreadActivity.kt

```diff
@@ -0,0 +1,580 @@
+                lifecycleScope.launch {
+                    val result = routeLineAPI.setRoutes(
+                        listOf(RouteLine(result.routes.first(), null))
+                    )
+                    val style = mapboxMap.getStyle()
+                    if (style != null) {
+                        routeLineView.renderRouteDrawData(style, result)
+                    }
+                }
+
+                // update the camera position to account for the new route
+                viewportDataSource.onRouteChanged(result.routes.first())
+                viewportDataSource.evaluate()
+            } else {
+                // remove the route line and route arrow from the map
+                val style = mapboxMap.getStyle()
+                if (style != null) {
+                    routeLineAPI.clearRouteLine { value ->
+                        routeLineView.renderClearRouteLineValue(
+                            style,
+                            value
+                        )
+                    }
+                    routeArrowView.render(style, routeArrowAPI.clearArrows())
+                }
+
+                // remove the route reference to change camera position
+                viewportDataSource.clearRouteData()
+                viewportDataSource.evaluate()
+            }
+        }
+    }
+
+    private val navigationSessionStateObserver = NavigationSessionStateObserver {
+        logD("NavigationSessionState=$it", LOG_CATEGORY)
+        logD("sessionId=${mapboxNavigation.getNavigationSessionState().sessionId}", LOG_CATEGORY)
+    }
+
+    private lateinit var sdkThread: HandlerThread
+    private lateinit var sdkHandler: Handler
+
+    @SuppressLint("MissingPermission")
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = LayoutActivityHandlerThreadBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        mapboxMap = binding.mapView.getMapboxMap()
+
+        // initialize the location puck
+        binding.mapView.location.apply {
+            this.locationPuck = LocationPuck2D(
+                bearingImage = ContextCompat.getDrawable(
+                    this@HandlerThreadActivity,
+                    R.drawable.mapbox_navigation_puck_icon
+                )
+            )
+            setLocationProvider(navigationLocationProvider)
+            enabled = true
+        }
+
+        val waitForCreation = CountDownLatch(1)
+
+        sdkThread = HandlerThread("handler thread")
+        sdkThread.start()
+        sdkHandler = Handler(sdkThread.looper)
+        sdkHandler.post {
+            mapboxNavigation = MapboxNavigation(
+                NavigationOptions.Builder(this)
+                    .accessToken(getMapboxAccessTokenFromResources())
+                    .build()
+            )
+            waitForCreation.countDown()
+        }
+        waitForCreation.await()
```

Comment:
It may lead to ANRs. post will execute the tasks in the order they were added, do we really need to wait here until `MapboxNavigation` is created? Anyway it will only be accessed from tasks queued by `handler.post`, right?